src/DynamoCore/Utilities/LuceneSearchUtility.cs

```diff
@@ -50,28 +76,45 @@ public enum LuceneStorage
         // Holds the instance for the IndexSearcher
         internal IndexSearcher Searcher;
 
+        /// <summary>
+        /// Default constructor for LuceneSearchUtility, it will use the default storage type - RAM
+        /// </summary>
+        /// <param name="model"></param>
         internal LuceneSearchUtility(DynamoModel model)
         {
             dynamoModel = model;
+            startConfig = new LuceneStartConfig();
+        }
+
+        /// <summary>
+        /// Constructor for LuceneSearchUtility, it will use the storage type passed as parameter
+        /// </summary>
+        /// <param name="model"></param>
+        /// <param name="config"></param>
+        internal LuceneSearchUtility(DynamoModel model, LuceneStartConfig config)
+        {
+            dynamoModel = model;
+            startConfig = config;
+            if (DynamoModel.IsTestMode)
+            {
+                // If under test mode, switch to the default storage type - RAM
+                startConfig = new LuceneStartConfig();
+            }
         }
 
         /// <summary>
         /// Initialize Lucene config file writer.
         /// </summary>
-        internal void InitializeLuceneConfig(string dirName, LuceneStorage storageType = LuceneStorage.FILE_SYSTEM)
+        internal void InitializeLuceneConfig()
```

Comment:
so is this function not just called from the constructor because it needs to be controlled separately in tests? It's just kind of confusing to see it scattered around the code base.

Reply:
The InitializeLuceneConfig fn is called separately in 3 cases. First for regular nodes, second for package names and third for tests or Nodeautocomplete(using RAM based).

Reply:
I have updated to put this function inside the constructor 👍🏻